.github/actions/build-frontend/action.yml

```diff
@@ -0,0 +1,35 @@
+# action.yml
+name: 'Build Frontend'
+description: 'Build react website'
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: Use Node.js ${{ matrix.node-version }} with yarn
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+
+    - run: yarn install
+      working-directory: frontend-react
+      shell: bash
+
+    - name: Build release for production branch (conditional check)
+      if: github.ref == 'refs/heads/production'
+      working-directory: frontend-react
+      run: yarn build:production
+      shell: bash
+
+    - name: Build staging for non-production branch (conditional check)
+      if: github.ref != 'refs/heads/production'
+      working-directory: frontend-react
+      run: yarn build:staging
+      shell: bash
+
+    - name: Save Static Website
+      uses: actions/upload-artifact@v2
+      with:
+        name: static_website_react-${{ github.run_id }}
+        path: frontend-react/build
+        retention-days: 3
```

Comment:
Another thought: the retention for production vs staging vs just-a-pr could be different? (e.g. PR could be 1 day, staging could be 3, production could be 14?) Maybe not worth the extra work.

Reply:
While I agree with staggering, it doesn't help the GitHub workflows as the uploaded artifacts are only available to be downloaded by the same workflows:
[GitHub docs](https://github.com/github/docs/blob/main/content/actions/advanced-guides/storing-workflow-data-as-artifacts.md?plain=1#L54)

I believe this is a security measure. Of course when we publish releases, that will will be a bit different since it will only be production builds.